spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/BinderAwareChannelResolver.java

```diff
@@ -35,10 +40,12 @@
  * resolves the channel from the bean factory and, if not present, creates a new channel
  * and adds it to the factory after binding it to the binder. The binder is optionally
  * determined with a prefix preceding a colon.
+ *
  * @author Mark Fisher
  * @author Gary Russell
+ * @author Ilayaperumal Gopinathan
  */
-public class BinderAwareChannelResolver extends BeanFactoryMessageChannelDestinationResolver {
+public class BinderAwareChannelResolver extends BeanFactoryMessageChannelDestinationResolver implements SmartLifecycle {
```

Comment:
Here is the circular reference:

Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'binderAwareChannelResolver' defined in class path resource [org/springframework/cloud/stream/config/ChannelBindingServiceConfiguration.class]: Unsatisfied dependency expressed through constructor argument with index 0 of type [org.springframework.cloud.stream.binder.BinderFactory]: : Error creating bean with name 'binderFactory': Injection of autowired dependencies failed; nested exception is org.springframework.beans.factory.BeanCreationException: Could not autowire method: public void org.springframework.cloud.stream.binder.DefaultBinderFactory.setBindersHealthIndicator(org.springframework.boot.actuate.health.CompositeHealthIndicator); nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'org.springframework.cloud.stream.config.ChannelBindingAutoConfiguration': Injection of autowired dependencies failed; nested exception is org.springframework.beans.factory.BeanCreationException: Could not autowire field: private java.util.List org.springframework.cloud.stream.config.ChannelBindingAutoConfiguration.adapters; nested exception is org.springframework.beans.factory.BeanCurrentlyInCreationException: Error creating bean with name 'binderAwareChannelResolver': Requested bean is currently in creation: Is there an unresolvable circular reference?; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'binderFactory': Injection of autowired dependencies failed; nested exception is org.springframework.beans.factory.BeanCreationException: Could not autowire method: public void org.springframework.cloud.stream.binder.DefaultBinderFactory.setBindersHealthIndicator(org.springframework.boot.actuate.health.CompositeHealthIndicator); nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'org.springframework.cloud.stream.config.ChannelBindingAutoConfiguration': Injection of autowired dependencies failed; nested exception is org.springframework.beans.factory.BeanCreationException: Could not autowire field: private java.util.List org.springframework.cloud.stream.config.ChannelBindingAutoConfiguration.adapters; nested exception is org.springframework.beans.factory.BeanCurrentlyInCreationException: Error creating bean with name 'binderAwareChannelResolver': Requested bean is currently in creation: Is there an unresolvable circular reference?